### PR TITLE
Object.assign fallback (v2)

### PR DESCRIPTION
**Pull request by XaserAcheron**
New version of #142, more or less.

**This project uses [semver](http://semver.org), please check the scope of this pr:**

- [ ] #none# - documentation fixes and/or test additions
- [x] #patch# - backwards-compatible bug fix
- [ ] #minor# - adding functionality in a backwards-compatible manner
- [ ] #major# - incompatible API change

# CHANGELOG

- Adds a fallback for Object.assign for browsers that don't support it (i.e. IE).

### CODE REVIEW

**Comment by XaserAcheron** [comment]
I don't know why the linter is griping at this particular line. It works [here](https://github.com/ciena-blueplanet/ember-prop-types/blob/master/addon/utils/validators/index.js#L22) just fine.

**Comment by XaserAcheron** [comment]
Just pushed a new version that satisfies the linter. Should hopefully be good this time.

Side-thought: It's probably worth putting `eslint` and all the required plugins in `devDependencies` and add it to the `ember test` suite. I had to install a bunch of packages globally, one by one, until it booted up. That's no fun. :P

File: addon/mixins/prop-types.js [hunk 1]
```diff
@@ -2,11 +2,13 @@
  * The PropTypesMixin definition
  */
 import Ember from 'ember'
-const {Mixin, get, getWithDefault, typeOf} = Ember
+const {Mixin, assign, get, getWithDefault, merge, typeOf} = Ember
 import config from 'ember-get-config'
 
 import PropTypes, {getDef, logger, validators} from '../utils/prop-types'
 
+const objectAssign = Object.assign || assign || merge
```
Comment (krasnoukhov) [inline addon/mixins/prop-types.js]: I think they want to call it `assignIt`, see here:

https://github.com/juwara0/ember-prop-types/blob/ccf8844162af270f7d4aa9f07b973f2b76387d24/addon/utils/validators/index.js#L22

**Comment by juwara0** [comment]
👍

**Pull request merged**